From cbc3d52186da0f3de950b1fcd1628a31ef46c3d7 Mon Sep 17 00:00:00 2001
From: cpw <cpw@weeksfamily.ca>
Date: Mon, 12 Oct 2015 14:44:39 -0400
Subject: [PATCH 078/122] Fix IncludeEntireClasspath so it doesn't dump out
 random other files from the classpath. Also, it doesn't try and save
 undecompiled additional classes too.

---
 src/org/jetbrains/java/decompiler/main/Fernflower.java    | 3 ++-
 src/org/jetbrains/java/decompiler/struct/ContextUnit.java | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/org/jetbrains/java/decompiler/main/Fernflower.java b/src/org/jetbrains/java/decompiler/main/Fernflower.java
index 65733ac..6afe975 100644
--- a/src/org/jetbrains/java/decompiler/main/Fernflower.java
+++ b/src/org/jetbrains/java/decompiler/main/Fernflower.java
@@ -113,7 +113,8 @@ public class Fernflower implements IDecompiledData {
         if (found.contains(file.getAbsolutePath()))
           continue;
 
-        if (file.exists()) {
+        // only add .class files from classpath
+        if (file.exists() && (file.getName().endsWith(".class") || file.getName().endsWith(".jar"))) {
           DecompilerContext.getLogger().writeMessage("Adding File to context from classpath: " + file, Severity.INFO);
           structContext.addSpace(file, false);
           found.add(file.getAbsolutePath());
diff --git a/src/org/jetbrains/java/decompiler/struct/ContextUnit.java b/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
index 850c3d8..e304396 100644
--- a/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
+++ b/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
@@ -110,6 +110,7 @@ public class ContextUnit {
         // classes
         for (int i = 0; i < classes.size(); i++) {
           StructClass cl = classes.get(i);
+          if (!cl.isOwn()) continue;
           String entryName = decompiledData.getClassEntryName(cl, classEntries.get(i));
           if (entryName != null) {
             String content = decompiledData.getClassContent(cl);
-- 
2.21.0.windows.1

